engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoSource.java

```diff
@@ -1,44 +1,72 @@
 package org.enso.interpreter.runtime.data;
 
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.library.ExportLibrary;
+import com.oracle.truffle.api.library.ExportMessage;
 import com.oracle.truffle.api.source.Source;
+import org.enso.interpreter.runtime.EnsoContext;
+import org.enso.interpreter.runtime.data.text.Text;
+import org.enso.interpreter.runtime.data.vector.ArrayLikeHelpers;
 
 /** Wrapper for exposing sources to Enso. Delegates to original methods with no behavior changes. */
-public class EnsoSource {
+@ExportLibrary(InteropLibrary.class)
+public final class EnsoSource implements EnsoObject {
+  private static final String[] MEMBERS = {
+    "getLanguage", //
+    "getName", //
+    "getPath", //
+    "isInternal", //
+    "getCharacters", //
+    "getLength", //
+    "getLineCount", //
+  };
   private final Source source;
 
   public EnsoSource(Source source) {
     this.source = source;
   }
 
-  public String getLanguage() {
-    return source.getLanguage();
-  }
-
-  public String getName() {
-    return source.getName();
-  }
-
-  public String getPath() {
-    return source.getPath();
-  }
-
-  public boolean isInternal() {
-    return source.isInternal();
+  /*
```

Comment:
A leftover?

Reply:
No, this one (`getCharacters` for ` line`) isn't implemented, but looks like it doesn't matter.